contactlayout.php

```diff
@@ -137,12 +137,25 @@ function contactlayout_civicrm_pageRun(&$page) {
         foreach ($layout['blocks'] as &$row) {
           foreach ($row as &$column) {
             foreach ($column as &$block) {
-              if (!empty($block['profile_id'])) {
-                $relatedContact = ProfileRelatedContact::get($contactID, !empty($block['related_rel']) ? $block['related_rel'] : '');
-                $profileContact = $relatedContact ? $relatedContact : $contactID;
-                $profileBlocks[$block['profile_id']] = CRM_Contactlayout_Page_Inline_ProfileBlock::getProfileBlock($block['profile_id'], $profileContact);
+              if (empty($block['profile_id'])) {
+                continue;
+              }
+
+              $profileContact = $contactID;
+              $block['rel_cid'] = NULL;
+              $block['rel_isMissing'] = FALSE;
+
+              if (!empty($block['related_rel'])) {
+                $relatedContact = ProfileRelatedContact::get($contactID, $block['related_rel']);
+                $profileContact = $relatedContact;
                 $block['rel_cid'] = $relatedContact;
+                $block['rel_isMissing'] = $relatedContact === NULL;
               }
+
+              $profileBlocks[$block['profile_id']] = CRM_Contactlayout_Page_Inline_ProfileBlock::getProfileBlock(
+                $block['profile_id'],
+                $profileContact
```

Comment:
For instances when the $profileContact is NULL i.e when the related contact is NULL and the profile contact is assigned this contact, won't there be errors in this function as the function is expecting an integer?

Reply:
The code doesn't actually break or throw an error, but added a condition not to execute it if the relationship is not found. This also doesn't break the code and the "empty" block is shown as normal.

Reply:
Even when the profile block is not added, the template ill still be loaded here: https://github.com/compucorp/org.civicrm.contactlayout/blob/1d016a0eb05c1335ed586c6e38e78bbb3d28c2c6/templates/CRM/Contact/Page/View/SummaryHook.tpl#L13 but since you didn't notice any errors and it works fine, maybe smarty has a way of handling that.

Reply:
@tunbola the basic bloc info, like the path, title, etc. is still there. What's not added is the block field names and values. We still need a path to the template and the title to display the  no results message.

Reply:
True that. What I actually was concerned about was references to profiles in the template eventhough it won't exist e.g https://github.com/civicrm/org.civicrm.contactlayout/blob/4cce50d55c90266f25e29d47c970a5409c724b25/templates/CRM/Contactlayout/Page/Inline/Profile.tpl#L2